src/GetPdoExpressionHashMapCapableTrait.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use Dhii\Expression\ExpressionInterface;
+use Dhii\Expression\LiteralTermInterface;
+use Dhii\Expression\TermInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use InvalidArgumentException;
+
+/**
+ * Common functionality for objects that can generate an expression value hash map for use in PDO parameter binding.
+ *
+ * @since [*next-version*]
+ */
+trait GetPdoExpressionHashMapCapableTrait
+{
+    /**
+     * Retrieves the expression value hash map for a given SQL condition, for use in PDO parameter binding.
+     *
+     * @since [*next-version*]
+     *
+     * @param TermInterface         $condition    The condition instance.
+     * @param string[]|Stringable[] $ignore       A list of term names to ignore, typically column names.
+     * @param array                 $valueHashMap The value hash map reference to write to.
+     *
+     * @return array A map of value names to their respective hashes.
+     */
+    protected function _getPdoExpressionHashMap(TermInterface $condition, array $ignore = [], array &$valueHashMap = [])
+    {
+        if ($valueHashMap === null) {
+            $valueHashMap = [];
+        }
+
+        if ($condition instanceof LiteralTermInterface) {
+            $value = $condition->getValue();
+            $value = $this->_normalizeString($value);
```

Comment:
Undocumented throw?

Reply:
Also, looks like a case for `OutOfRangeException`.

Reply:
Yep! Agree.